stea/stea_config.py

```diff
@@ -1,171 +1,107 @@
-import datetime
+from datetime import datetime, date
+from typing import Dict, Optional
 
-import configsuite
-from configsuite import MetaKeys as MK
-from configsuite import types
+from pydantic import BaseModel, conlist, field_validator, Field, model_validator
+from typing_extensions import Self
 
+from .stea_keys import SteaKeys
 
-@configsuite.validator_msg("Must be defined")
-def _is_not_empty(container):
-    return len(container) > 0
 
+def replace_dashes(string: str) -> str:
+    return string.replace("-", "_")
 
-@configsuite.transformation_msg("Fix key names by replacing: `-` with `_`")
-def _fix_keys(elem):
-    fix_dict = {}
-    for key in elem:
-        new_key = str(key).replace("-", "_")
-        if key != new_key:
-            print(f"Warning: Replacing deprecated {key} with {new_key}")
-        fix_dict[new_key] = elem.get(key)
-    return fix_dict
 
+class SimulatorProfile(BaseModel):
+    ecl_key: str = Field(description="Summary key", alias_generator=replace_dashes)
+    start_date: Optional[date] = Field(
+        None,
+        description=(
+            "By default fmu-steaclient will calculate a profile from the full "
+            "time range of the simulated data, but you can optionally use the keywords "
+            "start_date and end_year to limit the time range. Production on the "
+            "start_date is included. Date format is YYYY-MM-DD."
+        ),
+    )
+    start_year: Optional[int] = Field(None, description="Deprecated. Use start_date")
```

Comment:
The visibility of this deprecation warning changes a bit with this. I guess this is what we really want: https://github.com/pydantic/pydantic/pull/8237